From be169b45503bdaedc173924226d70523657779bc Mon Sep 17 00:00:00 2001
From: Helge Deller <deller@gmx.de>
Date: Sat, 6 Jan 2024 20:02:10 +0100
Subject: [PATCH 24/27] parisc: Fix PDC_MEM_MAP for OpenBSD 7.4
Forwarded: yes

The PDC_MEM_MAP call used find_hppa_device_by_hpa() instead
of find_hppa_device_by_path() and as such did not find the device
at a given path.
This fixes boot of OpenBSD 7.4.

Noticed-by: Bruno Haible <bruno@clisp.org>
Signed-off-by: Helge Deller <deller@gmx.de>
---
 src/parisc/parisc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/roms/seabios-hppa/src/parisc/parisc.c b/roms/seabios-hppa/src/parisc/parisc.c
index 5f450f78..17aee3b2 100644
--- a/roms/seabios-hppa/src/parisc/parisc.c
+++ b/roms/seabios-hppa/src/parisc/parisc.c
@@ -2032,7 +2032,7 @@ static int pdc_mem_map(unsigned int *arg)
     switch (option) {
         case PDC_MEM_MAP_HPA:
             dprintf(0, "\nSeaBIOS: PDC_MEM_MAP_HPA  bus = %d,  mod = %d\n", dp->path.bc[4], dp->path.mod);
-            dev = find_hppa_device_by_hpa(memmap->hpa);
+            dev = find_hppa_device_by_path(dp, NULL, 1);
             if (!dev)
                 return PDC_NE_MOD;
             memcpy(memmap, dev->mod_info, sizeof(*memmap));
-- 
2.39.2

